packages/xo-server-perf-alert/src/index.js

```diff
@@ -589,6 +589,7 @@ ${monitorBodies.join('\n')}`
 
       const entriesWithMissingStats = []
       for (const entry of snapshot) {
+        if (entry.object.physical_size === -1) continue
```

Comment:
`physical_size` does not exist.

As per the discussion, you need to handle both `-1` and `0` (simply `<= 0`).

Reply:
I printed the ```entry``` object and ```physical_size``` is a key 🤔 
```bash
{
  uuid: '6e6ee285-0efe-63d8-c354-d5f760e4d96c',
  object: SR {
    uuid: '6e6ee285-0efe-63d8-c354-d5f760e4d96c',
    name_label: 'XCP-ng Tools',
    name_description: 'XCP-ng Tools ISOs',
    allowed_operations: [
      'unplug',
      'plug',
      'pbd_create',
      'pbd_destroy',
      'vdi_clone',
      'scan'
    ],
    current_operations: {},
    VDIs: [
      'OpaqueRef:ed8d2e90-2bf9-46fc-81be-d171bba583f5',
      'OpaqueRef:dc3ea9f9-1984-4021-8f55-83a947067666',
      'OpaqueRef:c9dc5557-6733-4b41-b031-c7aa1206a1d1',
      'OpaqueRef:b2a8b729-70f4-41a8-aa32-ce49507e67a4',
      'OpaqueRef:279fc935-c62d-4732-83d1-0a15705f41bc'
    ],
    PBDs: [
      'OpaqueRef:c9256067-2561-413d-8eb5-9855672bb6c6',
      'OpaqueRef:b55e7791-cc61-4cac-8f07-97bb018fdca5',
      'OpaqueRef:09c359b4-d9ee-4cfb-8aa7-4061a1d61651'
    ],
    virtual_allocation: 47398912,
    physical_utilisation: -1,
    physical_size: -1,
    type: 'iso',
    content_type: 'iso',
    shared: true,
    other_config: {
      xensource_internal: 'true',
      xenserver_tools_sr: 'true',
      'i18n-key': 'xenserver-tools',
      'i18n-original-value-name_label': 'XCP-ng Tools',
      'i18n-original-value-name_description': 'XCP-ng Tools ISOs'
    },
    tags: [],
    sm_config: {},
    blobs: {},
    local_cache_enabled: false,
    introduced_by: 'OpaqueRef:NULL',
    clustered: false,
    is_tools_sr: true
  },
  objectLink: '[XCP-ng Tools](undefined#/srs/6e6ee285-0efe-63d8-c354-d5f760e4d96c/general)',
  value: 100,
  shouldAlarm: true,
  threshold: 80,
  observationPeriod: 60,
  listItem: '  * [XCP-ng Tools](undefined#/srs/6e6ee285-0efe-63d8-c354-d5f760e4d96c/general): 100.0% used\n'
}
```

Reply:
Ok, my bad, I thought you were handling XO objects here, but you are handling XAPI objects instead :+1: